src/main/pacman/pacman_agent.py

```diff
@@ -0,0 +1,350 @@
+        for i in range(len(self.enemies)):
+            if self.player == self.enemies[i]:
+                reward = -self.ENEMY_PENALTY
+        if reward == 0:
+            if self.player == self.food:
+                reward = self.FOOD_REWARD
+            else:
+                reward = -self.MOVE_PENALTY
+
+        if reward == self.FOOD_REWARD or reward == -self.ENEMY_PENALTY or self.episode_step >= self.MAX_ITERS:
+            done = True
+        else:
+            done = False
+
+        return new_observation, reward, done
+
+    def get_image(self): # Returns an image of the game board
+        env = np.zeros((self.SIZE, self.SIZE, 3), dtype=np.uint8)  # starts an rbg of our size
+        env[self.food.x][self.food.y] = self.FOOD_COLOR  # sets the food location tile to green color
+        env[self.player.x][self.player.y] = self.PLAYER_COLOR  # sets the player tile to blue
+        for enemy in self.enemies:
+            env[enemy.x][enemy.y] = self.ENEMY_COLOR  # sets the enemy location to red
+        for wall in self.walls:
+            env[wall.x][wall.y] = self.WALL_COLOR  # sets the player tile to blue
+        img = Image.fromarray(env, 'RGB')  # reading to rgb. Apparently. Even tho color definitions are bgr. ???
+        return img
+
+    def render(self):
+        img = self.get_image()
+        img = img.resize((300, 300))  # resizing so we can see our agent in all its glory.
+        cv2.imshow("image", np.float32(img))
+        cv2.waitKey(1)
+
+    def plotRewards(self, episode_rewards):
+        moving_avg = np.convolve(episode_rewards, np.ones((self.SHOW_EVERY,)) / self.SHOW_EVERY, mode='valid')
+
+        plt.plot([i for i in range(len(moving_avg))], moving_avg)
+        plt.ylabel(f"Reward {self.SHOW_EVERY}ma")
+        plt.xlabel("episode #")
+        plt.show()
+
+class PacmanPlayer:
+    # Definitions related to the student RL agent: model, update function
+    def __init__(self, update_after = 'each', OBSERVATION_SPACE_SIZE = (10, 10, 3), ACTION_SPACE_SIZE = 8,
+                 ep_number = 0, RENDER_EVERY = 10000, DISCOUNT = 0.99, BATCH_SIZE = 1000): #Update after: All = all subtasks; Each = each subtask
+
+        self.update_after = update_after
+        self.OBSERVATION_SPACE_SIZE = (10, 10, 3)
+        self.ACTION_SPACE_SIZE = 8
+        self.ep_number = 0
+        self.RENDER_EVERY = 1000 # render every 1000 episodes
+        self.DISCOUNT = 0.99
+        self.BATCH_SIZE = 1000
+
+        self.model = self.create_model()
+
+        # Custom tensorboard object
+        # self.tensorboard = ModifiedTensorBoard(log_dir="logs/{}-{}".format(MODEL_NAME, int(time.time())))
+
+    def create_model(self):
+        model = Sequential()
+
+        model.add(Conv2D(256, (3, 3),
+                         input_shape=self.OBSERVATION_SPACE_SIZE))  # OBSERVATION_SPACE_VALUES = (10, 10, 3) a 10x10 RGB image.
+        model.add(Activation('relu'))
+        model.add(MaxPooling2D(pool_size=(2, 2)))
```

Comment:
Now that feature map size is less you may not need pooling at all. Anyway this won't change anything in compute speed.